src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -2678,3 +2678,40 @@ def test_loop_var_exception(parser):
         "Loop-variable name 'i' is not declared and there are no unqualified "
         "use statements. This is currently unsupported."
         in str(excinfo.value))
+
+
+def test_named_and_wildcard_use_var(f2008_parser):
+    ''' Check that we handle the case where a variable is accessed first by
+    a wildcard import and then by a named import. '''
+    reader = FortranStringReader('''
+        module test_mod
+          use some_mod
+        contains
+          subroutine test_sub1()
+            ! a_var here must be being brought into scope by the
+            ! `use some_mod` in the module.
+            a_var = 1.0
+          end subroutine test_sub1
+          subroutine test_sub2()
+            use some_mod, only: a_var
+            a_var = 2.0
+          end subroutine test_sub2
+        end module test_mod
+        ''')
+    prog = f2008_parser(reader)
+    psy = PSyFactory(api="nemo").create(prog)
+    # We should have an entry for "a_var" in the Container symbol table
+    # due to the access in "test_sub1".
+    container = psy.invokes.container
+    avar1 = container.symbol_table.lookup("a_var")
+    assert type(avar1) == Symbol
```

Comment:
pylint asks you to use isinstance instead
psyir/frontend/fparser2_test.py:2707:11: C0123: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)

Reply:
Yes, but pylint is wrong in this case as I want to check that it really is a Symbol and not a subclass. I've disabled the warning. (Sergi and I discussed this when I made the change to create Symbols instead of DataSymbols for unknown symbols. We could introduce a new UnknownSymbol class instead.)

Reply:
OK, thanks for explaining. There should probably be a comment in the code to that effect, but seeing as it's test code it's not that important.